pkg/provisioner/ironic/ironic.go

```diff
@@ -1289,6 +1289,15 @@ func (p *ironicProvisioner) Provision(hostConf provisioner.HostConfigData) (resu
 }
 
 func (p *ironicProvisioner) setMaintenanceFlag(ironicNode *nodes.Node, value bool) (result provisioner.Result, err error) {
+	if ironicNode == nil {
```

Comment:
There no conceivable scenario in which we'd get to this point without having an actual ironic node.

Reply:
Code uses ironicNode.UUID in call just below, so if something goes wrong and there is no check for `nil` application would panic.

Reply:
Could also help in tests

Reply:
It probably _should_ panic, because that's always going to be a bug.

Reply:
I would like to avoid calling `panic`, the whole reason of this line was actually to avoid panic in runtime when  `ironicNode.UUID` accessed and the `ironicNode` is undefined.

Reply:
Think of it as an assertion, so if somebody writes some obviously wrong code we will hear about it as soon as possible (preferably in unit tests).